### ISSUE TITLE
NIOSSLCertificate silently drops intermediate certificates when loading a PEM from memory

### ISSUE THREAD

**Issue by kevints** [apple/swift-nio-ssl#121]
I have a .pem file that contains the certificate chain my server will present (a leaf certificate and an intermediate CA certificate). However, when I load it from the filesystem myself the NIOSSLCertificate API doesn't parse the full chain. Worse, it silently drops the intermediary certificate.
```swift
// Load the certificate chain as a Data from somewhere
let pemEncodedCertificateChain = try FileManager.default.contents(atPath: "./chain.pem")
let certificate = try NIOSSLCertificate(buffer: .init(pemEncoded), format: .pem)
let source: NIOSSLCertificateSource = .certificate(certificate)
let config = TLSConfiguration.forServer(certificateChain: [source], ...)
// This appears to work, but now my server won't present its intermediate CA certificate
```

But when we allow NIOSSL to load the file itself it reads the whole chain out (which is what we want). The cardinality of certificateChain is wrong though, since chain.pem contains 2 certificates.
```swift
// Let NIO load the certificate chain from the filesystem itself
let config = TLSConfiguration.forServer(certificateChain: [.file("./chain.pem")], ...)
```

It seems like there should be a way to get the second behavior with a buffer I loaded myself (`chain.pem` might not be on the filesystem).

**Comment by Lukasa** [comment]
Agreed, this API needs to be dramatically cleaned up. We can emulate the logic BoringSSL uses to load multiple certs ourselves, and then we should probably retire this API and force users to turn their files in to `NIOSSLCertificate` objects through our own helper methods.

**Comment by Lukasa** [comment]
Fixed by #123.
